src/development/platform-integration/android/restore-state-android.md

```diff
@@ -0,0 +1,212 @@
+this object _unless_ the scope has two (or more) buckets with
+the same `restorationId`. If that occurs,
+create a new `RestorationScope` to ensure
+that each ID is unique to its scope.
+Each scope is assigned a unique [`restorationScopeId`][].
+
+You must decide what state you want to save and restore,
+but you can only save specific types to a `RestorationBucket`,
+namely: `null`, `bool`, `int`, `double`, `String`, `Uint8List`
+(and other typed data), `List`, `Map`, and child `RestorationBucket`s.
+
+{{site.alert.note}}
+  The code in this page was taken from [VeggieSeasons][].
+  VeggieSeasons is a sample app written for iOS that uses Cupertino
+  widgets. An iOS app requires [a bit of extra setup][] in Xcode, but
+  the restoration classes otherwise work the same on both iOS and Android.
+{{site.alert.end}}
+
+
+[a bit of extra setup]: {{site.api}}/flutter/services/RestorationManager-class.html#state-restoration-on-ios
+[`restorationId`]: {{site.api}}/flutter/widgets/RestorationScope/restorationId.html
+[`restorationScopeId`]: {{site.api}}/flutter/widgets/RestorationScope/restorationScopeId.html
+[`RestorationScope`]: {{site.api}}/flutter/widgets/RestorationScope-class.html
+[VeggieSeasons]: {{site.github}}/flutter/samples/tree/master/veggieseasons
+
+## Step 1: Enabling state restoration
+
+Most every Flutter app contains a `main` method that calls the
+[`runApp`][] function. You can use `runApp`
+to set up the framework and the bindings that bind
+the Dart code to the native Flutter engine.
+Set up the `RestorationManager` in the `runApp` function.
+For example:
+
+```dart
+void main() {
+  /...
+
+  runApp(
+    const RootRestorationScope(
+      restorationId: 'root',
+      child: VeggieApp(),
+    ),
+  );
+}
+```
+
+QUESTION for Michael: I thought that Flutter automatically created
+  the Scope for every route. But this app manually
+  creats a Scope for the page. ?? (xxx)
+
+[`runApp`]: {{site.api}}/flutter/widgets/runApp.html
+
+## Step 2: Add RestorationMixin to the State class
+
+To implement restorable state,
+add the [`RestorationMixin`][] to the app's `State` class.
+
+```dart
+class _VeggieAppState extends State<VeggieApp> with RestorationMixin
+```
+
+Once you've added the restore state mixin,
+your IDE asks you to implement an abstract method
+called [`restoreState`][]. The `restoreState` method calls
+[`registerForRestoration`][] to register that you
+intend to save state. For example:
+
+```dart
+@override
+void restoreState(RestorationBucket? oldBucket, bool initialRestore) {
+  registerForRestoration(_appState, 'state');
+} 
+```
+
+[`registerForRestoration`]: {{site.api}}/flutter/widgets/RestorationMixin/registerForRestoration.html
+[`RestorationMixin`]: {{site.api}}/flutter/widgets/RestorationMixin-mixin.html
+[`restoreState`]: {{site.api}}/flutter/widgets/RestorationMixin/restoreState.html
+
+## Step 3: Define a restorationScopeId for each scope
+
+Each Scope in the app should have a `restorationScopeId`,
```

Comment:
Providing a restorationScopeId or restorationId to a widget turns on state restoration for that particular widget: It will now store its own state under the provided it in the surrounding restoration bucket. (the reference to route doesn't make much sense here, IMO). The difference between a restorationId and a restorationScopeID is that widgets that take a restorationScopeID will create a new restorationScope (i.e. RestoraionBucket) into which all children will store their state. A restorationId means the widget (and its children) just store the data in the surrounding bucket.